docs/msi_v2/slc_revocation_spec.md

```diff
@@ -49,123 +49,150 @@ sequenceDiagram
+| **AADSTS1000610–14 Auto-Remediation**           | eSTS returns 401 `invalid_client` with any of 1000610–1000614.                                          | MSAL calls `/issuecredential?bypass_cache=true`, obtains a new certificate, retries; success or deterministic failure surfaced. |
+| **Unspecified Credential Issue**                 | eSTS returns `invalid_client` without codes.                                                            | MSAL forces new certificate (bypass cache) and retries. |
+| **Claims Challenge Path**                        | Resource returns 401 with claims; app passes claims to MSAL.                                            | MSAL mints new certificate (`bypass_cache=true`), acquires token **with claims**. |
+| **IMDS/IssueCredential Failure Path**            | `/issuecredential` fails.                                                                               | MSAL returns failure with diagnostics; no infinite retry. |
+| **Telemetry Validation**                         | Verify counter/tags capture AADSTS code, `bypass_cache=true`, and `CredentialOutcome`.                  | Telemetry populated as specified. |
 
 ## Client-Side Telemetry
 
-To improve observability and diagnostics of Managed Identity (MSI) scenarios within MSAL, we propose introducing a **new telemetry counter** named `MsalMsiCounter`. This counter will be incremented (or otherwise recorded) whenever MSI token acquisition activities occur, capturing the most relevant context in the form of tags.
+To improve observability and diagnostics of Managed Identity (MSI) scenarios within MSAL, we use a **telemetry counter** named `MsalMsiCounter`.
 
 ### Counter Name
 - **`MsalMsiCounter`**
 
 ### Tags
-Each time we increment `MsalMsiCounter`, we include the following tags:
-
-1. **MsiSource**  
-   Describes which MSI path or resource is used.  
-   - Possible values: `"AppService"`, `"CloudShell"`, `"AzureArc"`, `"ImdsV1"`, `"ImdsV2"`, `"ServiceFabric"`
-
-2. **TokenType**  
-   Specifies the type of token being requested or used.  
-   - Possible values: `"Bearer"`, `"POP"`, `"mtls_pop"`
-
-3. **bypassCache**  
-   Indicates whether the MSAL cache was intentionally bypassed.  
-   - Possible values: `"true"`, `"false"`
-
-4. **CertType**  
-   Identifies which certificate was used during the MSI V2 flow.  
-   - Possible values: `"Platform"`, `"inMemory"`, `"UserProvided"`
-
-5. **CredentialOutcome**  
-   If using the `/credential` endpoint (ImdsV2) log the outcome.  
-   - Not found
-   - Retry Failed
-   - Retry Succeeded
-   - Success
-
-6. **MsalVersion**  
-   The MSAL library version in use.  
-   - Example: `"4.51.2"`
-
-7. **Platform**  
-   The runtime/OS environment.  
-   - Examples: `"net6.0-linux"`, `"net472-windows"`
+Each time we record `MsalMsiCounter`, include:
```

Comment:
At this point in time, I do not feel that the client telemetry is significant. We can add a feature request for this, but I would not prioritize it today.

Is there any server telemetry that we want to capture?